### PR TITLE
#142 MUI migration -- forms

### LINKED ISSUE #142: MUI Migration
## Description

Switch the frontend from using React-Bootstrap to the MUI component library. The goal is to maintain the look/functionality of the app as closely as possible while making the switch. 

For this ticket to be closed React-Bootstrap should be removed as a dependency and the changes should all be cleared with product.

Resources:
https://mui.com/

#### COMPONENTS/LAYOUT

- [x] #385
- [x] #386

#### PAGES

- [x] #388
- [x] #389
- [x] #390
- [x] #391
- [x] #392
- [x] #393
- [x] #394
- [x] #395
- [x] #396
- [x] #397
- [x] #398
- [x] #399
- [x] #400
- [x] #401
- [x] #402

### To Do as of 10/04/2022

- [x] Nav bar from above
- [x] #468
- [ ] #469
- [x] #470

### PR DESCRIPTION
**Pull request by jamescd18**
## Changes

Migrate a shit ton of forms to MUI

## Notes

I think I folded in LocalizationProvider, I forget

## Screenshots (if applicable)

I should add

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://github.com/Northeastern-Electric-Racing/FinishLine/blob/develop/docs/ContributorGuide.md) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors
- [x] No newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (if applicable)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] PR is linked to the ticket
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack

Closes #142

### CODE REVIEW

File: src/frontend/src/components/EditableTextInputList.tsx [hunk 1]
```diff
@@ -88,14 +91,18 @@ const EditableTextInputList: React.FC<EditableTextInputListProps> = ({
...
             type="text"
-            ref={isLastElement(index) ? focusRef : null}
+            id={`bullet-${index}`}
+            name={`bullet-${index}`}
             value={item.toString()}
-            placeholder={'Input new bullet here...'}
+            label="Work Package Name"
+            placeholder="Input new bullet here..."
+            ref={isLastElement(index) ? focusRef : null}
+            autoFocus={hasTyped && isLastElement(index)}
```
Comment (RChandler234) [inline src/frontend/src/components/EditableTextInputList.tsx:105]: Is the idea here that the last element in the list autofocuses? Or is this just to focus the last element in the list while you're typing in it?

**Comment by RChandler234** [comment]
@jamescd18 is this how your createCRpage looks rn 
image

**Comment by RChandler234** [comment]
The code is honestly too dense for me to nitpick rn, I'm gonna do a general overview of it to make sure everything looks fine, and just put all the bugs/ visual stuff I find here while clicking around

**Comment by RChandler234** [comment]
For Create New Change Request View:

- The input boxes need an outline when they're not clicked into/ would preferably have a slightly different color from the rest of the box (like it is in the app now)
- A dropdown is showing up for the wbsnum with recently entered numbers. Pretty sure this is a thing chrome generates, there should be a way to block it from showing up (I've seen it before). It's causing the background of that input to be blue when I select from it, so I say we disable it
- The Timeline Impact and Budget Impact bug out when you delete the number preset in there and click out of the input box.
- This new create CR page looks distinct enough from the old one that I feel like we should pull in Anthony/ Leroy and ppl to give feedback on design, because some elements feel a lil weird spacing wise to me
- The title moving up when you click in is cool, but imo we shouldn't have it unless it works everywhere without buginess (e.g. timeline/budget impact)

File: src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx [hunk 2]
```diff
@@ -72,26 +77,27 @@ const UpcomingDeadlines: React.FC = () => {
...
+          <InputLabel id="dateRange">Date Range</InputLabel>
+          <Select
+            label="Date Range"
+            labelId="dateRange"
             value={daysUntilDeadline}
             onChange={(e) => setDaysUntilDeadline(e.target.value)}
+            startAdornment={<InputAdornment position="start">Next</InputAdornment>}
+            endAdornment={
+              <InputAdornment position="end" sx={{ marginRight: 2 }}>
+                days
```
Comment (RChandler234) [inline src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx:90]: Capitalize Days

**Comment by RChandler234** [comment]
The spacing is a little weird here (too much on the right side), other than that and the other comment I left, upcoming deadlines looks good.
image

**Comment by RChandler234** [comment]
For Create New Project View:

- The input boxes need an outline when they're not clicked into/ would preferably have a slightly different color from the rest of the box (like it is in the app now)
- Same dropdown chrome thing I mentioned on the CR page happening on project name

**Comment by RChandler234** [comment]
I like the login page, no real notes on that, just looks cool

**Comment by RChandler234** [comment]
For EditableTextInputList:

- Looks good, I would see how the input boxes look with an outline even when they're not clicked into (doesn't look bad rn, since it's a list)

**Comment by RChandler234** [comment]
For Create WP View:

- The input boxes need an outline when they're not clicked into/ would preferably have a slightly different color from the rest of the box (like it is in the app now)
- The dropdown thing I mentioned on create new CR is happening on WP name input and Project WBS num input. I have now remembered, it has something to do with how you name the id, so maybe try changing that?
- Otherwise it looks good; just a opinion thing for the date picker, I'd either have them type it in or use the datepicker pop-up and disable the other option, but up to you tbh

File: src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx (hunk 2 above)
Comment (jamescd18) [inline src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx:90]:
```suggestion
                Days
```

File: src/frontend/src/components/EditableTextInputList.tsx (hunk 1 above)
Comment (jamescd18) [reply src/frontend/src/components/EditableTextInputList.tsx:105]: yep, I didn't code this, I think Anthony did.

**Comment by RChandler234** [comment]
The only goofy thing rn is the error states on the new CR page, you'll see if you try to submit without entering a field, for some reason this form functions differently than the other two for error handling (you can see p quickly by testing one of the two other forms in a similar way). Gonna look at it more when I have time, but overall this is ready for you to take a look @anthonybernardi

**Pull request merged**